### PR TITLE
Add init container declaration

### LINKED ISSUE #1294: Modify the request/limit for all containers that are in the Sidecar
**Is your feature request related to a problem? Please describe.**
On the cluster that I am working, we have limitranges configured and the min request is 256Mi.

When I try to make the okteto up I receive and error, because I have the issue as I said with the limitranges.

**Describe the solution you'd like**
We able to inject some configuration in order to increase/modify the request/limit.

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: jLopezbarb <javier@okteto.com>

Fixes #1294 

## Proposed changes
- Add initContainer to the manifest declaration
-

### CODE REVIEW

File: pkg/model/dev.go [hunk 1]
```diff
@@ -202,6 +203,12 @@ type PersistentVolumeInfo struct {
 	Size         string `json:"size,omitempty" yaml:"size,omitempty"`
 }
 
+// InitContainer represents the initial container
+type InitContainer struct {
```
Comment (rberrelleza) [inline pkg/model/dev.go:207]: What about also specifying the user/group? Or will this inherit the one from the main config?

File: pkg/k8s/deployments/translate.go [hunk 2]
```diff
@@ -219,6 +220,29 @@ func TranslateDevContainer(c *apiv1.Container, rule *model.TranslationRule) {
 	TranslateContainerSecurityContext(c, rule.SecurityContext)
 }
 
+func TranslateInitContainer(initContainer *model.InitContainer) {
+	if initContainer.Image == "" {
+		initContainer.Image = "okteto/bin:1.2.21"
```
Comment (rberrelleza) [inline pkg/k8s/deployments/translate.go:225]: Can we put this in a const, so it's easier to find/change when we update the tag?

**Review by rberrelleza — COMMENTED** [review]
Can we add tests for parsing the manifests, and to validate that `okteto init` doesn't create this section on the generated manifests?

File: pkg/k8s/deployments/translate.go [hunk 3]
```diff
@@ -219,6 +220,29 @@ func TranslateDevContainer(c *apiv1.Container, rule *model.TranslationRule) {
 	TranslateContainerSecurityContext(c, rule.SecurityContext)
 }
 
+func TranslateInitContainer(initContainer *model.InitContainer) {
+	if initContainer.Image == "" {
```
Comment (rberrelleza) [inline pkg/k8s/deployments/translate.go:224]: What's going to happen if the init container provided by the user doesn't have the binaries we need?  Can we check that we fail gracefully in that scenario?

File: pkg/model/dev.go [hunk 4]
```diff
@@ -202,6 +203,12 @@ type PersistentVolumeInfo struct {
...
+	Image     string               `json:"image,omitempty" yaml:"image,omitempty"`
```
Comment (rberrelleza) [inline pkg/model/dev.go:208]: For the usecase where the user is going to push our image into their private repository, how are we going to communicate to the user which bin image we're using?

File: pkg/k8s/deployments/translate.go (hunk 3 above)
Comment (rberrelleza) [reply pkg/k8s/deployments/translate.go:224]: Also, what happens if the user has an older bin image here? How will we troubleshoot that?

File: pkg/model/dev.go (hunk 1 above)
Comment (rberrelleza) [inline pkg/model/dev.go:207]: Would it possible to include this config in the `okteto doctor` file?

File: pkg/k8s/deployments/translate.go (hunk 2 above)
Comment (rberrelleza) [reply pkg/k8s/deployments/translate.go:225]: we might even want to do the same we do for the Version, and make it injectable via our CI/CD process.

File: pkg/k8s/deployments/translate.go (hunk 3 above)
Comment (pchico83) [reply pkg/k8s/deployments/translate.go:224]: we can add a binary to the `okteto/bin` image that returns the version of the bin image, and validate the version as part of the clean command:
https://github.com/okteto/okteto/blob/master/cmd/up/up.go#L899

File: pkg/model/dev.go (hunk 1 above)
Comment (pchico83) [reply pkg/model/dev.go:207]: No need to specify the user group

Comment (pchico83) [reply pkg/model/dev.go:207]: It is already done, we include the okteto manifest

File: pkg/model/dev.go (hunk 4 above)
Comment (pchico83) [reply pkg/model/dev.go:208]: `okteto up` will fail with a clear message

**Review by pchico83 — COMMENTED** [review]
@jLopezbarb let's add logic to validate that the user is using the right `okteto/bin` image

File: pkg/k8s/deployments/translate.go (hunk 2 above)
Comment (pchico83) [reply pkg/k8s/deployments/translate.go:225]: @jLopezbarb can we use `model. OktetoBinImageTag` here?

File: images/bin/Makefile [hunk 5]
```diff
@@ -2,4 +2,4 @@ tag = 1.2.21
 
 .PHONY: push
 push:
-	okteto build -t okteto/bin:${tag} .
+	okteto build -t jlopezbarb/bin:${tag} --build-arg OKTETO_BIN_TAG=${tag} .
```
Comment (pchico83) [inline images/bin/Makefile:5]:
```suggestion
	okteto build -t okteto/bin:${tag} --build-arg OKTETO_BIN_TAG=okteto/bin:${tag} .
```

File: cmd/up/up.go [hunk 6]
```diff
@@ -422,6 +425,11 @@ func (up *upContext) activate(autoDeploy, build bool) error {
 			}
 		}
 
+		if version != model.GetCurrentImageTagVersion() {
```
Comment (pchico83) [inline cmd/up/up.go:428]:
```suggestion
		if version != model.OktetoBinImage {
```

File: cmd/up/up.go [hunk 7]
```diff
@@ -422,6 +425,11 @@ func (up *upContext) activate(autoDeploy, build bool) error {
...
+			log.Yellow("The version of the binary image provided does not match the one used in this version.")
```
Comment (pchico83) [inline cmd/up/up.go:429]: The Okteto CLI version {version} uses the init container image { model.OktetoBinImage }.

File: cmd/up/up.go [hunk 8]
```diff
@@ -422,6 +425,11 @@ func (up *upContext) activate(autoDeploy, build bool) error {
...
+			log.Yellow("Please consider upgrading your binary.")
```
Comment (pchico83) [inline cmd/up/up.go:430]: Please consider upgrading your init container image { dev.InitContainer.Image} with the content of  { model.OktetoBinImage }

**Review by pchico83 — APPROVED** [review]
Minor comments, but good to go!

**Review by pchico83 — APPROVED** [review]

**Pull request merged**